code/modules/events/blowout.dm

```diff
@@ -107,7 +104,8 @@
 
 			for_by_tcl(A, /obj/machinery/door/airlock)
 				if (A.z != Z_LEVEL_STATION)
-					break
+					continue
 				if (!(istype(A, /obj/machinery/door/airlock/maintenance) || istype(A, /obj/machinery/door/airlock/pyro/maintenance) || istype(A, /obj/machinery/door/airlock/gannets/maintenance) || istype(A, /obj/machinery/door/airlock/gannets/glass/maintenance)))
 					continue
-				A.req_access = list(access_maint_tunnels)
+				if (isnull(A.req_access))
+					A.req_access = list(access_maint_tunnels)
```

Comment:
why not use initial here - this will give doors that had null access maint tunnel access required (exterior airlocks)

Reply:
Could potentially still lose access if a foot had it granted through access_spawn objects